ui/projects/Deploy.qml

```diff
@@ -0,0 +1,85 @@
+import QtQuick 2.12
+import QtQuick.Controls 2.12
+
+import "../../backend/sccs" as Backend
+import "../../backend/core"
+import "../common"
+
+Item {
+    id: root
+
+    property alias environment: backend.environment
+    property alias repositoryName: backend.repositoryName
+    property alias version: backend.version
+
+    Backend.RepoTriggerContinuousDeployment {
+        id: backend
+
+        onSuccess: {
+
+            for (var j = 0; j < Store.currentProject.length; j++) {
+                if (Store.currentProject[j].envName === root.environment) {
+                    for (var z= 0; z < Store.currentProject[j].repositories.length; z++) {
+
+                        if ( Store.currentProject[j].repositories[z].name === repositoryName) {
+                            Store.currentProject[j].repositories[z].version = root.version;
+                            Store.currentProject[j].repositories[z].isUpdated = true;
+                        }
+                    }
+                }
+
+            }
+
+            Store.currentProjectChanged();
+        }
+
+        onErrorChanged: {
+            if(isError()) {
+                console.log("An error occured: " + error)
+            }
+        }
+    }
+
+    BusyIndicator {
+        id: processing
+        visible: backend.processing
+        running: visible
+
+        width: 50
+        height: 50
+    }
+
+    Button {
+        visible: {
+
+            var isVisible = !backend.processing && backend.environment !== "none";
+
+            if (isVisible) {
+                for (var j = 0; j < Store.currentProject.length; j++) {
```

Comment:
This is a duplicate of code to find a repository in the currentProject based on environment and repositoryName.
You should consider to remove this duplicate code to create a find function reusable.
That will permit you to fix the fact that you are looping on all repositories and env despite you already find what you are looking for (exactly as done on another file).

in the current code, you are looping on all repositories and env despite the only thing that you want is to find a specific repository repositoryName in a specific environment root.enviroment to check if the button needs to be invisible.

Reply:
It look like the other loop but this is not a duplicate. I could improve this in anothor version.